astropy/io/ascii/ui.py

```diff
@@ -347,6 +348,13 @@ def read(table, guess=None, **kwargs):
             reader = get_reader(**new_kwargs)
             format = reader._format_name
 
+        if isinstance(table, (str, bytes, os.PathLike)):
```

Comment:
@svank - At this point `table` might be a Gb-long string, so it may be best to also apply the test in the core reading functionality that a file name cannot have `\n` or `\r` in it. This restriction is documented in the io.ascii docs.  So something like below:
```
    if isinstance(table, (str, bytes, os.PathLike)):
        # Don't expand a common case where table is a data string. File names
        # for io.ascii cannot have newlines in them and io.ascii does not accept
        # bytes data.
        is_data = isinstance(table, str) and ('\n' in table or '\r' in table):
        if not is_data:
            ex_user = os.path.expanduser(table)
            if os.path.exists(ex_user):
                table = ex_user
```